test/metabase/test/data/impl.clj

```diff
@@ -152,9 +153,15 @@
         ;; code may run inside of some other block that sets report timezone
         ;;
         ;; require/resolve used here to avoid circular refs
-        ((requiring-resolve 'metabase.test.util/do-with-temporary-setting-value)
-         :report-timezone nil
-         #(create-database! driver dbdef)))))))
+        (letfn [(thunk []
+                  (binding [api/*current-user-id*              nil
+                            api/*current-user-permissions-set* nil]
+                    (create-database! driver dbdef)))]
```

Comment:
Make sure current user isn't bound when creating a test database, otherwise permissions can cause sync to fail